PhpUnit/ContainerBuilderHasServiceDefinitionConstraint.php

```diff
@@ -12,16 +12,11 @@ class ContainerBuilderHasServiceDefinitionConstraint extends Constraint
     private $expectedClass;
     private $checkExpectedClass;
 
-    public function __construct($serviceId, $expectedClass = null, $checkExpectedClass = true)
-    {
-        if (!is_string($serviceId)) {
-            throw new \InvalidArgumentException('The $serviceId argument should be a string');
-        }
-
-        if ($checkExpectedClass && !is_string($expectedClass)) {
-            throw new \InvalidArgumentException('The $expectedClass argument should be a string');
-        }
-
+    public function __construct(
+        string $serviceId,
+        ?string $expectedClass = null,
+        $checkExpectedClass = true
```

Comment:
bool ?

Reply:
Ok, I will add `bool` here, and also dive into the rabbit hole of adding less obvious type hints.

Reply:
Ḩ̶͚̗͙̻̝͖̠̹̊́̂̓̆̀͑͋́͆͊̚͝ë̷̦̪́́͂̎̓̇̈́̃̐̑͝ṟ̵̨̲͚͚̻͎̫͉̊é̴̹͎̺̺͔̤̇̅͐̂͗̊͠͠͝ ̴̨̨̬̗̥̥̟̙̣͚͌̀͂̎̃͋̆̿̎͒́̄̾̐͝ì̸͚͚̲̜͈͈̓t̶͔̿͆̅̇͌̒̑̎͊ ̴͖̹͚̮̍̓̈̓́ͅḉ̶̛̩̙̙̽̐̇̇͝͝ͅo̷̡͎̙͑͆̆͊̋̎̋͋̚͝͝͝m̷̧̨̖̱͔̺̼͎̘̒͗̾̆̔̀́̿̇̚ȇ̶͎͝s̸̢̤͓͚͔̹̏́̈́̏͆͂!̵͓̯͚̮̩̻̯̱̻̥̝̭͊̐͐̾̚͠ ̴̘̻̼̰̫͙̇͌͂̋͆͜͝T̸̢̲͔̼̹͛̃̉̉̓̉̀̔͐̎̑͊̕͘͝h̸̤̭̱͎̼̺̟͇̭͚̤͌͌͛̿̎̚ͅͅè̵̛̳͉̖̤̥̬ ̵̢̧̗̝̻̥̲͙̱̫͓̣̪̮̂̽̈́͜h̵̲̭̱͖͊͂̓̓́͘u̵̹̔̀́͐̍̈͘̕g̷͔̳͇͙͔̩̩̙̊e̷͎͚̱͇͆̔̉̎͂͆̇͆̉͋̓͗̔͝ ̴̧͙̪̙̯̏͂́̈p̶̧̘̯͖̼̖̼̓̇͐̉͐̃̊̅̓̅̏̕̚u̸͓̓̋̿̿̑͝ļ̸̰͖̮̬͔̝̳͉̘͔̍̈́́̐̄̈̋̀̌̔̈́͘͝l̴̯̭͌̆́̊̓̐̓ ̸̧̨̹̝̠̤̜̜̘̑̓͂r̸̢͍̘͖̗̻͋̓͛̓̃̇̓̄͋͗́͘͘͠e̵͓̠̟̬̞̎́͊̂̾͘q̵̛͓̊̇̊͆̊̎͗̓͜ṳ̸̹̼͉̭͇̬̹͇͉̏̇̆̿̀̈́̏͐̈͘ḙ̸̢̢̨̡̛̜̖̪̤̤̫̣͉͍̅̎̀̔͒̎̊̽͆͐̕͠ͅs̴̨̡̛̬͓̘̰̗̗̒̀́t̷̡̡̡͖͕͙̥͈̩̲̱̳̪̉̑̃͆̅́͒̿́̈́̚͝!̷̤̙̣̣͎̗̤̟͕̅͆̀̓͛̊̈ͅͅ ̷̬̒͐̏̍̃͆͂̀̓̋̑̕͜B̷̧͇̑̆̀̀̎̈́̋̃͑͐͋͊̀̕e̸͈͈͙̼͕̳̞͈͈͉̙͋͂̽̄̉͛̉̓̇̐͑͘͠h̶̨̢̡̙̜͙͍͈̫̹͌̂ô̸̤͊l̴̢̡̨͓͖̱̪̝̪̗͗̇͒̈̐̾̈́̍̅͆͠͝d̸̢̬̺̻̜̣̉͛̓!̵̢̢̳͉̱͉͉̰̗̪͍̜̺̲̈́̓̾͐̐̉̄̑̈́